docs/TSG.md

```diff
@@ -282,8 +282,37 @@ Since this flame was essentially all of CPU 4, whatever is taking the most signi
 > TODO
 
 ## Why is Performance bad across all my Connections?
+1. [UDP receive offload is not working.](#diagnosing-udp-receive-offload-issues-windows-only)
 
-1. [The work load isn't spreading evenly across cores.](#diagnosing-rss-issues)
+2. [The work load isn't spreading evenly across cores.](#diagnosing-rss-issues)
+
+### Diagnosing UDP Receive Offload Issues (Windows only)
+
+UDP Receive Offload (URO) is an importance performance feature. To check if URO is working correctly, you can collect TCPIP ETW traces and look for URO related events. To collect a TCPIP ETW trace, run the below commands and convert the resulting ETL file to text file using `netsh trace convert <etl_file_path>`. [Here's](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-r2-and-2012/jj129382(v=ws.11)) a MSDN doc for collecting network trace in Windows.
```

Comment:
Do we have an MSDN article on the topic? If so, it'd be great to link `UDP Receive Offload` to that article.

Reply:
URO is still not documented externally yet.

Reply:
IMO, we should fix that ASAP.

Reply:
I agree but I will leave this task to NickG.